app/src/main/java/xyz/block/gosling/MainActivity.kt

```diff
@@ -61,10 +62,19 @@ class MainActivity : ComponentActivity() {
         settingsManager = SettingsManager(this)
         isAccessibilityEnabled = settingsManager.isAccessibilityEnabled
 
-        // Start the Agent service
-        startForegroundService(Intent(this, Agent::class.java))
-
-        startService(Intent(this, OverlayService::class.java))
+        // Check for overlay permission
+        if (!Settings.canDrawOverlays(this)) {
+            // If not granted, request it
+            val intent = Intent(
+                Settings.ACTION_MANAGE_OVERLAY_PERMISSION,
+                Uri.parse("package:$packageName")
+            )
+            startActivity(intent)
+        } else {
+            // Start services only if we have overlay permission
+            startForegroundService(Intent(this, Agent::class.java))
+            startService(Intent(this, OverlayService::class.java))
+        }
```

Comment:
If we request permission here, do we not start it, i.e. only the next time? I'd expect something that we would try again when the request comes back

Reply:
further down it will start it in that case (I think??)

Reply:
(ie if it bounces them out, then it will resume later?) or am I wrong?

Reply:
yeah - added some logging, it does seem to call resume if you bounce out and back